elixir/raindrops/lib/raindrops.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Raindrops do
+  @doc """
+  Returns a string based on raindrop factors.
+
+  - If the number contains 3 as a prime factor, output 'Pling'.
+  - If the number contains 5 as a prime factor, output 'Plang'.
+  - If the number contains 7 as a prime factor, output 'Plong'.
+  - If the number does not contain 3, 5, or 7 as a prime factor,
+    just pass the number's digits straight through.
+  """
+
+  @prime_factors [3, 5, 7, 1]
+  @sounds %{3 => "Pling", 5 => "Plang", 7 => "Plong", 1 => ""}
+
+  @spec convert(pos_integer) :: String.t()
+  def convert(number) do
+    case get_sound(number) do
+      "" -> to_string(number)
+      sound -> sound
+    end
+  end
+
+  def get_sound(number) do
+    Enum.reduce(@prime_factors, "", fn(x, acc) ->
+      case rem(number, 2) do
```

Comment:
What is the purpose of rem? I don't get this.

Reply:
Yeah, that is a good question. And the answer is: this is me writing code in a rush at 6 am in the morning. the second argument in rem does of course not make any sense, it needs to be x